no-octal rule on NonOctalDecimalIntegerLiterals

<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).

    This template is for bug reports. If you are here for another reason, please see below:

    1. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To request a change that is not a bug fix, rule change, or new rule: https://eslint.org/docs/developer-

guide/contributing/changes
    4. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:**
5.16.0
* **Node Version:**
10.16.0
* **npm Version:**
6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2018,
  },
  rules: {
    'no-octal': 'error',
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
let x;
x = 018;
x = 081;
x = 018.2;
x = 081.2;
x = 018e2;
x = 081e2;
```

<!-- Paste the command you used to run ESLint: -->
```bash
eslint index.js
```

**What did you expect to happen?**
By the [specification](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-additional-syntax-numeric-literals) both 018 and 081 are NonOctalDecimalIntegerLiteral, not LegacyOctalIntegerLiteral.

If the no-octal rule is supposed to catch these literals (which look like octal literals
but are actually evaluated as decimal literals), then it should report all 6 as errors. 

Otherwise, if the rule is supposed to catch only octal literals, then it should report zero errors, and perhaps a new rule could be added.

**What actually happened? Please include the actual, raw output from ESLint.**
Only 018, 018.2 and 018e2 are reported as errors:

  2:5  error  Octal literals should not be used  no-octal
  4:5  error  Octal literals should not be used  no-octal
  6:5  error  Octal literals should not be used  no-octal

✖ 3 problems (3 errors, 0 warnings)



**Are you willing to submit a pull request to fix this bug?**
yes

--- a/lib/rules/no-octal.js
+++ b/lib/rules/no-octal.js
@@ -28,7 +28,7 @@ module.exports = {
         return {
 
             Literal(node) {
-                if (typeof node.value === "number" && /^0[0-7]/u.test(node.raw)) {
+                if (typeof node.value === "number" && /^0[0-9]/u.test(node.raw)) {
                     context.report({ node, message: "Octal literals should not be used." });
                 }
             }

--- a/tests/lib/rules/no-octal.js
+++ b/tests/lib/rules/no-octal.js
@@ -30,6 +30,14 @@ ruleTester.run("no-octal", rule, {
     invalid: [
         { code: "var a = 01234;", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
         { code: "a = 1 + 01234;", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
-        { code: "00", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] }
+        { code: "00", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "08", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "09.1", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "09e1", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "09.1e1", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "018", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "019.1", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "019e1", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] },
+        { code: "019.1e1", errors: [{ message: "Octal literals should not be used.", type: "Literal" }] }
     ]
 });

